### PR TITLE
Update task servicetest to move dependency to the new TaskControlService

### LINKED ISSUE #12724: Refactor TestTaskService framework to be more flexible.
servicetest.go is designed to test anything that implements the TaskService interface. It currently relies on having a task.Store. To make this system more flexible we should update it to depend on the TaskControlService instead. This may require a test adaptor to be created to adapt the store and the logWriter to the new functions.

### PR DESCRIPTION
**Pull request by lyondhill**
closes #12724

We will now have the capability to write new task services that dont have to implement the backend.Store or LogReader or LogWriters

### CODE REVIEW

File: task/servicetest/servicetest.go [hunk 1]
```diff
@@ -37,17 +37,109 @@ func init() {
...
+	r, err := tcs.lr.FindRunByID(ctx, st.Org, runID)
+	if err == nil {
+		schedFor, _ = time.Parse(time.RFC3339, r.ScheduledFor)
+		reqAt, _ = time.Parse(time.RFC3339, r.RequestedAt)
+	}
+
+	// in the old system the log store may not have the run until after the first
+	// state update, so we will need to pull the currently running.
+	tt := time.Time{}
+	if schedFor == tt {
```
Comment (mark-rushakoff) [inline task/servicetest/servicetest.go]: Prefer `if schedFor.IsZero()`.

File: task/servicetest/servicetest.go [hunk 2]
```diff
@@ -37,17 +37,109 @@ func init() {
...
+		for _, cr := range m.CurrentlyRunning {
+			if influxdb.ID(cr.RunID) == runID {
+				schedFor = time.Unix(cr.Now, 0)
+				reqAt = time.Unix(cr.RequestedAt, 0)
+			}
+		}
+
+	}
+
+	// Update the run state to Started; normally the scheduler would do this.
```
Comment (mark-rushakoff) [inline task/servicetest/servicetest.go]: Comment is stale. Fine to delete it.

File: task/servicetest/servicetest.go [hunk 3]
```diff
@@ -37,17 +37,109 @@ func init() {
...
+	}
+
+	r, err := tcs.lr.FindRunByID(ctx, st.Org, runID)
+	if err != nil {
+		return err
+	}
+	schFor, _ := time.Parse(time.RFC3339, r.ScheduledFor)
+	reqAt, _ := time.Parse(time.RFC3339, r.RequestedAt)
+
+	// Update the run state to Started; normally the scheduler would do this.
```
Comment (mark-rushakoff) [inline task/servicetest/servicetest.go]: Delete comment.

File: task/servicetest/servicetest.go (hunk 1 above)
Comment (lyondhill) [reply task/servicetest/servicetest.go]: Oo yah I knew that was somewhere. Thanks.

**Pull request merged**